### ISSUE TITLE
Widget Package Does Not Support React v18

### ISSUE THREAD

**Issue by akarys92** [Uniswap/widgets#19]
**Bug Description**
The widget does not support the current version of React (v18). To get it to work you have to downgrade projects to React 17. When I try to run `npm install --save @uniswap/widgets` on a React v18 project I get the following error: 

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: broken-widget@0.1.0
npm ERR! Found: react@18.1.0
npm ERR! node_modules/react
npm ERR!   react@"^18.1.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^17.0.1" from @uniswap/widgets@1.0.10
npm ERR! node_modules/@uniswap/widgets
npm ERR!   @uniswap/widgets@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Steps to Reproduce**
1. Start a new react app project using Create React App v5 
`create-react-app my-app`
2. Note that the React version in `package.json` is v18
```
{
...
dependencies: {
       "react": "^18.1.0",
       "react-dom": "^18.1.0",
    }
    ...
}
```
3. Per the [docs](https://docs.uniswap.org/sdk/widgets/swap-widget) attempt to install Uniswap widget package:
```
npm install --save @uniswap/widgets react-redux
```
4. Note the error above is returned

**Expected Behavior**
The package should install without error and be usable per the [docs](https://docs.uniswap.org/sdk/widgets/swap-widget) description.

**Comment by kristiehuang** [comment]
for now, this worked well for me as a workaround fix: https://dev.to/lowla/create-react-app-with-a-previous-version-of-react-4g03

**Comment by akarys92** [comment]
Fixed by PR: #24
